packages/compiler/server/tmlanguage.ts

```diff
@@ -226,6 +226,17 @@ const tupleExpression: BeginEndRule = {
   patterns: [expression],
 };
 
+const arrayExpression: MatchRule = {
```

Comment:
Hmm, this is less general than "array expression", it doesn't match (A | B)[], for example. It would be rather hard to express the general concept in tmlanguage, but I checked and typescript does the same thing we do now with it. You get a separate tuple expression for the `[]`. In general I think it is fine to be (deliberately) imprecise in tmlanguage. It actually ends up giving a sort of error recovery when you allow more than the language. Since we already allow any expression after is/extends and treat array expression as an expression followed by a tuple expression, I think it would be fine to not do anything here. I was actually referring to the documentation of the language grammar in spec.emu.html when I asked for a grammar change. 😁